src/main/java/org/truffleruby/extra/TruffleGraalNodes.java

```diff
@@ -54,6 +54,16 @@ protected DynamicObject assertNotCompiled() {
 
     }
 
+    @CoreMethod(names = "assert_never_part_of_compilation", onSingleton = true)
+    public abstract static class AssertNeverPartOfCompilationNode extends CoreMethodArrayArgumentsNode {
+
+        @Specialization
+        protected DynamicObject assertNeverPartOfCompilation() {
+            throw new RaiseException(getContext(), coreExceptions().runtimeErrorNeverPartOfCompilation(this));
```

Comment:
Minor and not worth changing: I would have the message inline and not a method in CoreExceptions since it will only be used by one caller.